lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -111,12 +114,14 @@ def scrub(transcript)
 
       def add_extra_data(post, payment, options)
         post[:shopperEmail] = options[:shopper_email] if options[:shopper_email]
-        post[:shopperIP] = options[:shopper_ip] if options[:shopper_ip]
-        post[:shopperReference] = options[:shopper_reference] if options[:shopper_reference]
+        post[:shopperIP] = options[:ip] if options[:ip]
+        post[:shopperReference] = options[:customer_id] || options[:customer]
         post[:fraudOffset] = options[:fraud_offset] if options[:fraud_offset]
         post[:selectedBrand] = options[:selected_brand] || NETWORK_TOKENIZATION_CARD_SOURCE[payment.source.to_s] if payment.is_a?(NetworkTokenizationCreditCard)
         post[:deliveryDate] = options[:delivery_date] if options[:delivery_date]
         post[:merchantOrderReference] = options[:merchant_order_reference] if options[:merchant_order_reference]
+        post[:telephoneNumber] =  options[:phone] || (options[:billing_address][:phone] if options[:billing_address] &&
```

Comment:
What if `options[:phone]` is present but `options[:billing_address][:phone]` is not? Won't then `post[:telephoneNumber]` not be set?

Reply:
Also indicating a lack of tests

Reply:
Right, hadn't seen the parentheses. Why have the `if` at all then?